### ISSUE TITLE
Skipper S3 stream-upload memory buffer fills up while passing disk parameter 

### ISSUE THREAD

**Issue by super-iterator** [balderdashy/skipper-s3#42]
Trying to stream-upload to S3 using skipper and skipper-s3 using this code

```
image_upload : function  (req,res) {

		// depends on bodyparser:skipper in confing/http.js

		console.log("Uploading image .. ");

		req.file('image')
			.upload({
				adapter: require('skipper-s3'),
				key: KEY,
				secret: SECRET,
				bucket: 'mybucket',
                                tmpdir: 'tmp/s3-multiparts'

			}, function whenDone(err, uploadedFiles) {
				if (err){
					return res.negotiate(err);
				} else {
					dict=	{
						files: uploadedFiles,
						textParams: req.params.all()
					}
					return res.ok(dict);
				}
			}

		);
```

**I have 2 problems**
**First**: when I try to upload some file, I can see that this file buffers in memory, instead just buffering to the disk (path added to options).

Actually this is a big problem, as my memory will fill up so quickly when having many users concurrently uploading

**Second**: my application crashes when upload is bigger than 5MBs, giving me this error

```
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: Request aborted
    at IncomingMessage.onReqAborted (~Desktop/myApp/node_modules/sails/node_modules/skipper/node_modules/multiparty/index.js:175:17)
    at IncomingMessage.emit (events.js:104:17)
    at abortIncoming (_http_server.js:279:11)
    at Socket.serverSocketCloseListener (_http_server.js:292:5)
    at Socket.emit (events.js:129:20)
    at TCP.close (net.js:485:12)
```

What shall I do with those 2 problems?

**Comment by sgress454** [comment]
Hm, the tests for this are passing, meaning that for at least our test key and bucket, it's able to upload files >25mb without issue.  How are you determining that it's buffering to memory and not disk?  As for the error, unfortunately it's not giving you much info there, but you could try throwing a log around [here](https://github.com/andrewrk/node-multiparty/blob/master/index.js#L191) in the multiparty `index.js` file to see what the actual error is.

**Comment by super-iterator** [comment]
I run the sails app and run process monitor, which checks the disk, cpu, and memory utilization. Regarding throwing the error, will do in the close future

**Comment by sailsbot** [comment]
Thanks for posting, @super-iterator.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
